esphome/components/as7341/as7341.h

```diff
@@ -0,0 +1,144 @@
+#pragma once
+
+#include "esphome/core/component.h"
+#include "esphome/components/sensor/sensor.h"
+#include "esphome/components/i2c/i2c.h"
+
+namespace esphome {
+namespace as7341 {
+
+static const uint8_t AS7341_CHIP_ID = 0X09;
+
+static const uint8_t AS7341_CONFIG = 0x70;
+static const uint8_t AS7341_LED = 0x74;
+
+static const uint8_t AS7341_ENABLE = 0x80;
+static const uint8_t AS7341_ATIME = 0x81;
+
+static const uint8_t AS7341_WTIME = 0x83;
+
+static const uint8_t AS7341_AUXID = 0x90;
+static const uint8_t AS7341_REVID = 0x91;
+static const uint8_t AS7341_ID = 0x92;
+static const uint8_t AS7341_STATUS = 0x93;
+
+static const uint8_t AS7341_CH0_DATA_L = 0x95;
+static const uint8_t AS7341_CH0_DATA_H = 0x96;
+static const uint8_t AS7341_CH1_DATA_L = 0x97;
+static const uint8_t AS7341_CH1_DATA_H = 0x98;
+static const uint8_t AS7341_CH2_DATA_L = 0x99;
+static const uint8_t AS7341_CH2_DATA_H = 0x9A;
+static const uint8_t AS7341_CH3_DATA_L = 0x9B;
+static const uint8_t AS7341_CH3_DATA_H = 0x9C;
+static const uint8_t AS7341_CH4_DATA_L = 0x9D;
+static const uint8_t AS7341_CH4_DATA_H = 0x9E;
+static const uint8_t AS7341_CH5_DATA_L = 0x9F;
+static const uint8_t AS7341_CH5_DATA_H = 0xA0;
+
+static const uint8_t AS7341_STATUS2 = 0xA3;
+
+static const uint8_t AS7341_CFG1 = 0xAA;  ///< Controls ADC Gain
+
+static const uint8_t AS7341_CFG6 = 0xAF;  // Stores SMUX command
+static const uint8_t AS7341_CFG9 = 0xB2;  // Config for system interrupts (SMUX, Flicker detection)
+
+static const uint8_t AS7341_ASTEP = 0xCA;      // LSB
+static const uint8_t AS7341_ASTEP_MSB = 0xCB;  // MSB
+
+using as7341_adc_channel_t = enum {
+  AS7341_ADC_CHANNEL_0,
+  AS7341_ADC_CHANNEL_1,
+  AS7341_ADC_CHANNEL_2,
+  AS7341_ADC_CHANNEL_3,
+  AS7341_ADC_CHANNEL_4,
+  AS7341_ADC_CHANNEL_5,
+};
+
+using as7341_smux_cmd_t = enum {
+  AS7341_SMUX_CMD_ROM_RESET,  ///< ROM code initialization of SMUX
+  AS7341_SMUX_CMD_READ,       ///< Read SMUX configuration to RAM from SMUX chain
+  AS7341_SMUX_CMD_WRITE,      ///< Write SMUX configuration from RAM to SMUX chain
+};
+
+using as7341_gain_t = enum {
```

Comment:
Should be using this format to match the rest of ESPHome codebase =)
Same for the above enums
```suggestion
enum AS7341Gain {
```